sdk/communication/azure-communication-identity/tests/test_communication_identity_client_async.py

```diff
@@ -151,6 +176,42 @@ async def test_get_token_from_managed_identity(self):
         assert token_response.token is not None
 
     @recorded_by_proxy_async
+    async def test_get_token_from_managed_identity_voip(self):
+        if not is_live():
```

Comment:
180-188 lines are duplicated in each managed identity test case, can we get rid of this duplication?

Reply:
@AikoBB async test with proxy don't support table tests (parametrized).  I think it's because of issues with writing to a file system in parallel. Such tests were not supported before having a proxy, too

Reply:
can not we eliminate the duplication by introducing a method in the ACSIdentityTestCase which will provide the client?

Reply:
@AikoBB bu looking at all the code we're reusing, I came up with even more refactoring and now ACSIdentityTestCase has methods for returning async and sync clients both for connection string and managed identity.